perftestnotebook.py

```diff
@@ -144,66 +158,92 @@ def process(self):
 
         self.fmt_data = fmt_data
 
+        """
+        Write formatted data output to filepath
+        """
+        print(json.dumps(self.fmt_data, indent=4, sort_keys=True))
+
+        prefix = "output" if "prefix" not in self.config else self.config["prefix"]
+        output_data_filepath = "%s_fmt_data.json" % prefix
+
+        if "output" in self.config:
+            output_data_filepath = self.config["output"]
+
+        print("Writing results to %s" % output_data_filepath)
+
+        with open(output_data_filepath, "w") as f:
+            json.dump(self.fmt_data, f, indent=4, sort_keys=True)
+
+        """
+        Analyze data
+        """
         if "analysis" in self.config:
             # Analyze the data
-            all_results = {}
-            self.analyzer.data = fmt_data
-            for func in self.config["analysis"]:
-                all_results[func] = getattr(self.analyzer, func)()
-            return all_results
+            # TODO
```

Comment:
Undo this change, without this code, there's no way to run an analysis function.

The analyzer functions need to be changed to return code (in a string). Then, you need to take this returned code and insert it into a text file. This text file will then be uploaded as the "template file".